client/login/controller.js

```diff
@@ -224,312 +227,218 @@ export function qrCodeLogin( context, next ) {
-		}
-		if ( user_name ) {
-			authParams.append( 'user_name', user_name );
-		}
-
-		// Redirect to the destination URL with the auth token
-		const redirectUrl = redirectTo.includes( '?' )
-			? `${ redirectTo }&${ authParams.toString() }`
-			: `${ redirectTo }?${ authParams.toString() }`;
 
-		page.redirect( redirectUrl );
-		return true;
-	};
-
-	// If we have a response from Apple, handle it
-	if ( handleSocialResponseFromRedirect() ) {
-		return;
-	}
-
-	// If no response, initialize Apple auth
-	const initializeAppleAuth = async () => {
-		try {
-			// Get the redirect URI
-			const redirectUri = getRedirectUri();
-
-			// Generate a random oauth2State value like AppleLoginButton does
-			// We don't need to call the nonce endpoint as Apple implementation
-			// creates a random state locally
-			const oauth2State = String( Math.floor( Math.random() * 10e9 ) );
-			// Store state in session storage like AppleLoginButton does
-			window.sessionStorage.setItem( 'siwa_state', oauth2State );
-
-			// Create the state object with relevant data
-			const stateData = {
-				redirect_to: query?.redirect_to || '/',
-				is_jetpack: true,
-				locale: context.params.lang,
-				originalUrlPath: window.location.pathname,
-				queryString: window.location.search || null,
-			};
+			// Exchange auth code for tokens
+			const response = await postLoginRequest( 'exchange-social-auth-code', {
+				service: 'google',
+				auth_code: code,
+				redirect_uri: getRedirectUri(),
+				client_id: config( 'wpcom_signup_id' ),
+				client_secret: config( 'wpcom_signup_key' ),
+				state,
+			} );
 
-			// Use the same approach as AppleLoginButton - load the Apple JS SDK
-			const appleClientUrl =
-				'https://appleid.cdn-apple.com/appleauth/static/jsapi/appleid/1/en_US/appleid.auth.js';
+			const { access_token, id_token } = response.body.data;
 
-			await loadScript( appleClientUrl );
+			// Try to connect Google account to existing WordPress.com account
+			try {
+				// Prepare WPCOM API for authentication
+				require( 'wpcom-proxy-request' ).reloadProxy();
+				wpcom.req.post( { metaAPI: { accessAllUsersBlogs: true } } );
+
+				// Attempt to connect social account
+				const wpcomResponse = await wpcom.req.post( '/me/social-login/connect', {
+					service: 'google',
+					access_token,
+					id_token,
+					redirect_to: state.redirect_to,
+					client_id: config( 'wpcom_signup_id' ),
+					client_secret: config( 'wpcom_signup_key' ),
+				} );
+
+				// Use API-provided redirect if available
+				if ( wpcomResponse.redirect_to ) {
+					page.redirect( wpcomResponse.redirect_to );
+					return true;
+				}
 
-			if ( ! window.AppleID ) {
-				throw new Error( 'Failed to load Apple Sign In SDK' );
+				// Otherwise use default redirect with auth params
+				redirectWithAuthParams( access_token, id_token, state.redirect_to );
+				return true;
+			} catch ( connectError ) {
+				// If connection fails, try creating a new account
+				try {
```

Comment:
What do you think about moving this error handling (`try...catch`) to a separate function, so the code is easier to read?